instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/InstrumenterBuilder.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import io.opentelemetry.instrumentation.api.InstrumentationVersion;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class InstrumenterBuilder<REQUEST, RESPONSE> {
+  private final OpenTelemetry openTelemetry;
+  private final String instrumentationName;
+  private InstrumenterConstructor<REQUEST, RESPONSE> constructor =
```

Comment:
newlines between blocks of final uninitialized, final initialized, and non-final